src/main/java/org/openbase/bco/bcozy/util/Groups.java

```diff
@@ -1,35 +1,52 @@
 package org.openbase.bco.bcozy.util;
 
+import javafx.application.Platform;
 import javafx.collections.FXCollections;
 import javafx.collections.ObservableList;
 import javafx.util.StringConverter;
 import org.openbase.bco.registry.remote.Registries;
 import org.openbase.jul.exception.CouldNotPerformException;
+import org.openbase.jul.exception.printer.ExceptionPrinter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import rst.domotic.unit.UnitConfigType;
+import rst.domotic.unit.UnitTemplateType;
+import rst.domotic.unit.authorizationgroup.AuthorizationGroupConfigType;
 
 import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
 
 /**
  * @author vdasilva
  */
 public final class Groups {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(Groups.class);
+
     public static ObservableList<UnitConfigType.UnitConfig> getGroups() {
         ObservableList<UnitConfigType.UnitConfig> groups = FXCollections.observableArrayList();
 
-        try {
-            setGroups(Registries.getUserRegistry().getAuthorizationGroupConfigs(), groups);
-        } catch (CouldNotPerformException | InterruptedException ex) {
-            // not synchronized yet...
-            // ex.printStackTrace();
+        if (Registries.isDataAvailable()) {
+            try {
+                setGroups(Registries.getUserRegistry().getAuthorizationGroupConfigs(), groups);
+            } catch (CouldNotPerformException ex) {
+                ExceptionPrinter.printHistory(ex, LOGGER);
+            } catch (InterruptedException ex) {
+                Thread.currentThread().interrupt();
+            }
         }
 
         try {
             Registries.getUserRegistry().addDataObserver((observable, userRegistryData) ->
```

Comment:
I know this code already exists before but this is really inefficient. This observer is registered in every call of "getGroups()" but will never be removed again.

Reply:
I dont't know how else to do this better. 
The observer is needed throughout the whole runtime of bcozy because the list of groups can be updated any time so we need to listen for changes the whole time. Don't we?